pkg/core/options.go

```diff
@@ -17,6 +17,8 @@ type Settings struct {
 	profilingEnabled bool
 	memProfDir       string
 
+	label string
```

Comment:
Is this param here to indicate that we want to retrieve all versions of the listed?
Could it mean something like `wantsLabelVersions`?
This works, but why don't we simply add this as an ordinary parameter to the `List` functions? (I believe the change on the CLI pkg is pretty much localized - there is probably one single call).

Reply:
> Could it mean something like `wantsLabelVersions`?

yep, that's what this means.  i'm proceeding with "the same `struct` can have slightly different meanings in different places, and that's okay," idea.  note also, the

```
todo: validation method on settings (specificify settings type per use-case).
```

comment. i'm still slightly convinced that distinguishing the semantics of the `struct` by how and where its instances appear in the source is a good idea, although it's not the focus of this request.

----

about whether to use `... Options` -> `Settings` or not, i was actually thinking in the opposite direction.

what about passing what is currently the `prefix` parameter to `listLabelsChan` as one of the varadic `Option` parameters?

Reply:
ok. Either way makes sense. The only inconsistent situation is when we have some args via func (like the current `prefix`) and some args via opts (like your `label` aka `wantsVersions bool`). Either consistent approach is fine by me.

Reply:
ty for being permissive about prefs here.  i will push for consistency upon starting changes outside of `storage`.